binary_search_tree/binary_search_tree.py

```diff
@@ -3,20 +3,71 @@
 from dll_queue import Queue
 from dll_stack import Stack
 
+# Questions:
+# Only ints? 
+# Negative numbers?
+
+# Observations
+# >= goes right
+# Need to traverse to delete
+# When deleting, the smallest child becomes parent
+
+
 class BinarySearchTree:
   def __init__(self, value):
     self.value = value
     self.left = None
     self.right = None
+    self.root = None
+    self.currentRoot = None
+  # * `insert` adds the input value to the binary search tree, adhering to the
+  # rules of the ordering of elements in a binary search tree.
+  # Need to traverse to find spot to insert
 
+  # After creating a node we need to find out how to traverse the tree
+  # How do I know if I'm at the root of a tree?
+  # What is the root of a binary search tree really?
+  # Does it have 0,1, or 2 branches. A tree can only have one root and two branches
+  # But we don't know for example, if a root has one branch, we don't know if we should insert there. For example, we have (this is hard to describe on text), but a tree of 3-(l)2-(r)5 and we want to insert 4. So we start at three, four is greater than 3, so we go to five. Four is less than five, so we'll insert it there, unless, there's another element there.
+  # We have to initialize the tree with a value, so there'll always be a root initally.
   def insert(self, value):
-    pass
+    # When we create the tree, we always will have a root because the function call requires a value, which may become the root
+    if value < self.value:
+        self.left = value
```

Comment:
Think you've got a problem here, `self.left` (and `self.right`) needs to contain more than just the value for your tree to have more than 2 leaves